src/frontend/src/common/utils/dataTransforms.ts

```diff
@@ -0,0 +1,81 @@
+import { generateOrgSpecificUrl, ORG_API } from "src/common/api";
+import ENV from "src/common/constants/ENV";
+
+const { API_URL } = ENV;
+
+export interface IdMap<T> {
+  [key: string]: T;
+}
+
+/**
+ * Reduces an array of objects to a mapping between the keyString arg and the objects
+ * that make up the array. Effective for quickly looking up objects by id, for example.
+ */
+export const reduceObjectArrayToLookupDict = <T extends Record<string, any>>(
+  arr: T[],
+  keyedOn: string
+): IdMap<T> => {
+  const keyValuePairs = arr.map((obj: T) => {
+    const id = obj[keyedOn];
+    return [id, obj];
+  });
+  return Object.fromEntries(keyValuePairs);
+};
+
+/**
+ * Phylo run transforms
+ */
+
+/**
+ * If there is a tree associated with this run, return links to download tree data
+ */
+interface PhyloRunLinks {
+  downloadLinkIdStylePublicIdentifiers: string | undefined;
+  downloadLinkIdStylePrivateIdentifiers: string | undefined;
+  accessionsLink: string | undefined;
```

Comment:
Why not use `accessionsLink?: string`?  I'm assuming this is readability preference.  Maybe we should document stuff like this somewhere.

Reply:
If I don't allow undefined, then I can't start links off with anobject + keys that have explicitly `undefined` values.